doc/sphinx/addendum/rewrite-rules.rst

```diff
@@ -50,21 +49,23 @@ but they may still reduce using the provided rules, unlike axioms.
 
 Rewrite rules
 ---------------
+.. cmd:: Rewrite {| Rule | Rules } @ident := {? %| } {+| @rewrite_rule }
+  :name: Rewrite Rule; Rewrite Rules
 
   .. insertprodn rewrite_rule rewrite_rule
 
   .. prodn::
      rewrite_rule ::= {? @univ_decl %|- } @rw_pattern ==> @term
 
-Rewrite rules have two parts simply named left-hand side or pattern and right-hand side.
+Declares a named block of rewrite rules.
```

Comment:
What is the name for?

Currently it shares a namespace with constants/inductives (ie `Axiom foo : nat. Rewrite Rules foo := ...` is not allowed), is this necessary?

I guess the name is currently used in the "rewrite rules not supported here" errors, is that all?

Since there's a name it would be nice to be able to use Print on it, although currently I don't think we store enough info to print something user readable (mostly because variable names are dropped).

Reply:
The name is the only way I found to allow for rewrite rules to be included in the vo (through module entries).

They share a namespace with constants and inductives because I didn't think they deserved their own, even if they can't conflict with anything currently. (We may want something like `cbn [rules_set]` in the future, so this would be the correct namespace.)

I guess so, but I don't know if it's worth keeping these names and in what form.

Reply:
>We may want something like cbn [rules_set] in the future, so this would be the correct namespace

We could always have some separate syntax for that like `cbn [!rules]` but I guess it's ok.

The doc should mention that the names share that namespace IMO.